tests/RuleSet/RuleSetTest.php

```diff
@@ -419,23 +419,78 @@ public function testInvalidConfig(): void
         new RuleSet(['@Symfony:risky' => null]);
     }
 
-    private function sortNestedArray(array $array): array
+    private function sortNestedArray(array $array, string $ruleName): array
     {
-        foreach ($array as $key => $element) {
-            if (!\is_array($element)) {
-                continue;
-            }
-            $array[$key] = $this->sortNestedArray($element);
+        $this->doSort($array, $ruleName);
+
+        return $array;
+    }
+
+    /**
+     * Sorts an array of fixer definition recursively.
+     *
+     * Sometimes keys are all string, sometimes they are integers - we need to account for that.
+     *
+     * @param array<int|string,mixed> $data
+     */
+    private function doSort(array &$data, string $path): void
+    {
+        static $orderMatters = [
+            'ordered_imports.imports_order',
+            'phpdoc_order.order',
+        ];
+
+        if (\in_array($path, $orderMatters, true)) { // order matters
+            return;
         }
 
-        // sort by key if associative, by values otherwise
-        if (array_keys($array) === range(0, \count($array) - 1)) {
-            sort($array);
+        $keys = array_keys($data);
+
+        if ($this->allInteger($keys)) {
```

Comment:
Why this change?

Reply:
You mean this skipping of sorting an array of only arrays?
Ah, this change has strayed from a different branch. 
It shouldn't be here. At least yet.